File: apps/api/scripts/module-02-seed-query.py
from sqlalchemy import select

from app.db import SessionLocal
from app.models import Link


def main() -> None:
    code = "module02"
    long_url = "https://example.com/module-02"

    with SessionLocal() as db:
        existing = db.scalar(select(Link).where(Link.code == code))
        if existing is None:
            link = Link(code=code, long_url=long_url, created_by="module-02")
            db.add(link)
            db.commit()
        else:
            link = existing

        selected = db.scalar(select(Link).where(Link.code == code))
        if selected is None:
            raise RuntimeError("Expected inserted link to be found by code.")

        print(f"inserted code={link.code}")
        print(f"selected code={selected.code}")
        print(f"matched long_url={selected.long_url}")


if __name__ == "__main__":
    main()

File: scripts/module-02-seed-query.ts
import "dotenv/config";
import { PrismaPg } from "@prisma/adapter-pg";
import { PrismaClient } from "../generated/prisma";

const adapter = new PrismaPg({
  connectionString: process.env.DATABASE_URL!,
});

const prisma = new PrismaClient({ adapter });

async function main() {
  const uniqueCode = `short-${Math.random().toString(36).substring(2, 8)}`;
  const targetLongUrl = 'https://example.com';
  const testUser = 'user_testing_99';

 const result = await prisma.link.upsert({
  where: {
    longUrl_createdBy: {
      longUrl: targetLongUrl,
      createdBy: testUser,
    },
  },
  update: {},
  create: {
    code: uniqueCode,
    longUrl: targetLongUrl,
    createdBy: testUser,
  },
});

const selectedLink = await prisma.link.findUnique({
  where: {
    code: result.code,
  },
});

 
  

  // 3. Print output that includes the required fields
  
  if (selectedLink) {
    console.log(`inserted code: ${result.code}`);
    console.log(`selected code: ${selectedLink.code}`);
    console.log(`matched long_url: ${selectedLink.longUrl}`);
    
  } else {
    console.error('Failed to query the link back from the database.');
  }
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });